File: QuisineSpringBoot/Quisine/src/main/java/com/project/Quisine/entity/RestaurantFood.java
package com.project.Quisine.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.Data;

@Data
@Entity
public class RestaurantFood {

	@Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    
	@JsonIgnore
    @ManyToOne
    @JoinColumn(name = "restaurant_id")
    private UserEntity restaurant;
    
	@JsonIgnore
    @ManyToOne
    @JoinColumn(name = "food_id")
    private Food food;
    
    private float rate;
}

File: QuisineAngular/src/app/models/restaurant-food.ts
import { Food } from "./food";
import { Restaurant } from "./restaurant";

export class RestaurantFood {
    id: number;
  food: Food;
  restaurant: Restaurant;
  rate: number;
  addedToCart: boolean;
  quantity: number;

  constructor(id: number, food: Food, rate: number, restaurant: Restaurant) {
    this.id = id;
    this.food = food;
    this.restaurant = restaurant;
    this.rate = rate;
    this.addedToCart = false;
    this.quantity = 0;
  }
}
